oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -1123,7 +1127,20 @@ private Query createQuery(String propertyName, Filter.PropertyRestriction pr, Pr
         final String field = elasticIndexDefinition.getElasticKeyword(propertyName);
 
         if (pr.isNullRestriction()) {
-            return Query.of(q -> q.bool(b -> b.mustNot(m -> m.exists(e -> e.field(field)))));
+            // nullProps check has been added since 1.4.0. Use the old strategy when version is lower
+            if (elasticIndexDefinition.getMappingVersion().compareTo(MINIMUM_NULL_CHECK_VERSION) < 0) {
+                // check if the default mapping is >= 1.5.0
+                if (ElasticIndexDefinition.MAPPING_VERSION != null &&
+                        ElasticIndexDefinition.MAPPING_VERSION.compareTo(new ElasticSemVer(1, 5, 0)) >= 0) {
```

Comment:
Suggestion: create a constant with `new ElasticSemVer(1, 5, 0)`: avoids creating a new instance every time this logic is executed and improves readability because the name of the constant can document the meaning of this particular version (maybe `MAXIMUM_NULL_CHECK_VERSION`?)